consensus/istanbul/backend/announce.go

```diff
@@ -377,14 +396,49 @@ func (qed *queryEnodeData) DecodeRLP(s *rlp.Stream) error {
 	return nil
 }
 
+// getValProxyAssignments returns the remote validator -> external node assignments.
+// If this is a standalone validator, it will set the external node to itself.
+// If this is a proxied validator, it will set external node to the proxy's external node.
+func (sb *Backend) getValProxyAssignments(valAddresses []common.Address) (map[common.Address]*enode.Node, error) {
+	var valProxyAssignments map[common.Address]*enode.Node = make(map[common.Address]*enode.Node)
+	var selfEnode *enode.Node = sb.SelfNode()
+	var proxies map[common.Address]*proxy.Proxy // This var is only used if this is a proxied validator
+
+	for _, valAddress := range valAddresses {
+		var externalNode *enode.Node
+
+		if sb.IsProxiedValidator() {
+			if proxies == nil {
+				var err error
+				proxies, err = sb.proxiedValidatorEngine.GetValidatorProxyAssignments(nil)
+				if err != nil {
+					return nil, err
+				}
+			}
+			proxyObj := proxies[valAddress]
+			if proxyObj == nil {
+				continue
+			}
+
+			externalNode = proxyObj.ExternalNode()
+		} else {
+			externalNode = selfEnode
+		}
+
+		valProxyAssignments[valAddress] = externalNode
+	}
+
+	return valProxyAssignments, nil
+}
+
 // generateAndGossipAnnounce will generate the lastest announce msg from this node
 // and then broadcast it to it's peers, which should then gossip the announce msg
 // message throughout the p2p network if there has not been a message sent from
 // this node within the last announceGossipCooldownDuration.
 // Note that this function must ONLY be called by the announceThread.
 func (sb *Backend) generateAndGossipQueryEnode(version uint, enforceRetryBackoff bool) error {
 	logger := sb.logger.New("func", "generateAndGossipQueryEnode")
-	logger.Trace("generateAndGossipQueryEnode called")
+	logger.Trace("generateAndGossipQueryEnode called\n")
```

Comment:
Is the newline an artifact of a debugging session or is there a particular reason for it? If its just an artifact, please remove it.